core/src/validator/src/rules/duplicate-asset-name.ts

```diff
@@ -0,0 +1,55 @@
+import { BaseValidator } from "../core.ts";
+
+import { getStates } from "../utils/getState.ts";
+
+import type { Result } from "../utils/types.ts";
+import { metadataValidator } from "../utils/metadataChecks.ts";
+
+export class DuplicateAssetName extends BaseValidator {
+  constructor() {
+    const id = "duplicate-asset-name";
+    super(id);
+  }
+
+  async Execute(
+    asset_name: string,
```

Comment:
normalize variable name casing (here snake_case is used while camelCase is used everywhere else)